tests/framework/extensions/provisioning/creates.go

```diff
@@ -748,3 +754,174 @@ func createWindowsRegistrationCommand(command, publicIP, privateIP string, machi
 	}
 	return command
 }
+
+// AddRKE2K3SCustomClusterNodes is a helper method that will add nodes to the custom RKE2/K3S custom cluster.
+func AddRKE2K3SCustomClusterNodes(client *rancher.Client, cluster *v1.SteveAPIObject, nodes []*nodes.Node, rolesPerNode []string) error {
+	clusterStatus := &apiv1.ClusterStatus{}
+	err := v1.ConvertToK8sType(cluster.Status, clusterStatus)
+	if err != nil {
+		return err
+	}
+
+	token, err := tokenregistration.GetRegistrationToken(client, clusterStatus.ClusterName)
+	if err != nil {
+		return err
+	}
+
+	var command string
+	for key, node := range nodes {
+		logrus.Infof("Adding node %s to cluster %s", node.NodeID, cluster.Name)
+		if strings.Contains(rolesPerNode[key], "windows") {
+			command = fmt.Sprintf("powershell.exe %s -Address %s", token.InsecureWindowsNodeCommand, node.PublicIPAddress)
+		} else {
+			command = fmt.Sprintf("%s %s --address %s", token.InsecureNodeCommand, rolesPerNode[key], node.PublicIPAddress)
+		}
+
+		output, err := node.ExecuteCommand(command)
+		if err != nil {
+			return err
+		}
+
+		logrus.Infof(output)
+	}
+
+	err = kwait.Poll(500*time.Millisecond, defaults.TenMinuteTimeout, func() (done bool, err error) {
+		clusterResp, err := client.Steve.SteveType(clusters.ProvisioningSteveResourceType).ByID(cluster.ID)
+		if err != nil {
+			return false, err
+		}
+
+		if clusterResp.ObjectMeta.State.Name == active && nodestat.AllManagementNodeReady(client, cluster.ID, defaults.ThirtyMinuteTimeout) == nil {
+			return true, nil
+		}
+		return false, nil
+	})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// DeleteRKE2K3SCustomClusterNodes is a method that will delete nodes from the custom RKE2/K3S custom cluster.
+func DeleteRKE2K3SCustomClusterNodes(client *rancher.Client, clusterID string, cluster *v1.SteveAPIObject, nodesToDelete []*nodes.Node) error {
+	steveclient, err := client.Steve.ProxyDownstream(clusterID)
+	if err != nil {
+		return err
+	}
+
+	nodesSteveObjList, err := steveclient.SteveType("node").List(nil)
+	if err != nil {
+		return err
+	}
+
+	for _, nodeToDelete := range nodesToDelete {
+		for _, node := range nodesSteveObjList.Data {
+			if node.Annotations[internalIP] == nodeToDelete.PrivateIPAddress {
+				machine, err := client.Steve.SteveType(machineSteveResourceType).ByID(namespace + "/" + node.Annotations[machineNameAnnotation])
+				if err != nil {
+					return err
+				}
+
+				logrus.Infof("Deleting node %s from cluster %s", nodeToDelete.NodeID, cluster.Name)
+				err = client.Steve.SteveType(machineSteveResourceType).Delete(machine)
+				if err != nil {
+					return err
+				}
+
+				err = kwait.Poll(500*time.Millisecond, defaults.TenMinuteTimeout, func() (done bool, err error) {
+					_, err = client.Steve.SteveType(machineSteveResourceType).ByID(machine.ID)
+					if err != nil {
+						logrus.Infof("Node has successfully been deleted!")
+						return true, nil
+					}
+					return false, nil
+				})
+				if err != nil {
+					return nil
+				}
```

Comment:
```suggestion
				if err != nil {
					return err
				}
```

Reply:
Markus can you make a hot fix for the v2.7 branch?

Reply:
Yeah, gimme a few min

Reply:
Addressed